addons/website_crm_partner_assign/views/crm_lead_views.xml

```diff
@@ -7,27 +7,29 @@
             <field name="arch" type="xml">
                 <xpath expr="//notebook/page[last()]" position="after">
                     <page string="Assigned Partner" name="assigned_partner" groups="sales_team.group_sale_salesman">
-                        <group col="3">
-                            <label for="partner_latitude" string="Geolocation"/>
-                            <div class="o_row">
-                                <span class="oe_grey">( </span>
-                                <field name="partner_latitude"/>
-                                <span class="oe_grey" attrs="{'invisible':[('partner_latitude','&lt;=',0)]}">N </span>
-                                <span class="oe_grey" attrs="{'invisible':[('partner_latitude','&gt;=',0)]}">S </span>
-                                <field name="partner_longitude"/>
-                                <span class="oe_grey" attrs="{'invisible':[('partner_longitude','&lt;=',0)]}">E </span>
-                                <span class="oe_grey" attrs="{'invisible':[('partner_longitude','&gt;=',0)]}">W </span>
-                                <span class="oe_grey">) </span>
-                            </div>
-                            <div colspan="2">
-                                <button string="Automatic Assignment" name="action_assign_partner" type="object" class="btn-link"/>
-                            </div>
+                        <group>
```

Comment:
Ok I see what you mean but it's quite complicated in this case to not touch the dom, let me explain.
Here is what we want it to look like (15.3): https://tinyurl.com/2l5sze79
Because of the new grid layout, with only one group, each element is now displayed in their respective row: https://tinyurl.com/2h4wakvk 
The way to solve this, and what is done in every form view now, is having a parent group and 2 inner groups, one for the left and one for the right side of the form (what I did here): https://tinyurl.com/2kn9rxxn
If we don't want those nested groups, we could also try to display the `Automatic Assignment` and `Send Email` buttons directly in the row cell we want like this: https://tinyurl.com/2oe3r37o . However this involve using some divs with flex and width 50% to keep the correct display like so : 
```xml
<page string="Assigned Partner" name="assigned_partner" groups="sales_team.group_sale_salesman">
    <group>
        <label for="partner_latitude" string="Geolocation"/>
        <div class="d-flex">
            <div class="w-50">
                <span class="oe_grey">( </span>
                <field name="partner_latitude" class="oe_inline o_input_12ch px-1"/>
                <span class="oe_grey" attrs="{'invisible':[('partner_latitude','&lt;=',0)]}">N </span>
                <span class="oe_grey" attrs="{'invisible':[('partner_latitude','&gt;=',0)]}">S </span>
                <field name="partner_longitude" class="oe_inline o_input_12ch ps-2 pe-1"/>
                <span class="oe_grey" attrs="{'invisible':[('partner_longitude','&lt;=',0)]}">E </span>
                <span class="oe_grey" attrs="{'invisible':[('partner_longitude','&gt;=',0)]}">W </span>
                <span class="oe_grey ps-1">) </span>
            </div>
            <button string="Automatic Assignment" name="action_assign_partner" type="object" class="btn-link pt-0 ps-0"/>
        </div>

        <label for="partner_assigned_id"/>
        <div class="d-flex">
            <field class="w-50" name="partner_assigned_id" domain="[('grade_id','!=',False)]" context="{'partner_set_default_grade_activation': 1}"/>
            <button string="Send Email" type="action" class="btn-link pt-0 ps-0"
                attrs="{'invisible':[('partner_assigned_id','=',False)]}"
                name="%(crm_lead_forward_to_partner_act)d"
                context="{'default_composition_mode': 'forward','hide_forward_type': 1 , 'default_partner_ids': [partner_assigned_id]}"/>
        </div>
    </group>
</page>
```

In term of keeping the dom as close as possible as the existing one in 16, I think using nested groups will be less impacting because xpath like `//group/...` will still match the correct inner groups and just skip the parent group. However adding divs inside the structure makes it more complicated to match any xpath.

What do you think about it? (sorry for the long text)

Reply:
Thanks for the detailed explanations :) after that I tried various strange fixes, and no-one was really fixing the display issue. We effectively have to break the view ... so ok for your solution, while we break it better fix it in a clean way :)

This explanation could be included in the commit message, so that we know why we have to break the display :)

Cheers !